arch/xtensa/include/xtensa-asm2-s.h

```diff
@@ -306,7 +306,11 @@ _do_call_\@:
 	 * execution) while we muck with the windows and decrement the nested
 	 * count.  The restore will unmask them correctly.
 	 */
+#if XCHAL_HAVE_NMI
 	rsil a0, XCHAL_NMILEVEL
+#else
+	rsil a0, XCHAL_DEBUGLEVEL
```

Comment:
What if there is no debug level?

Reply:
we could make it simpler and have: `rsil a0, XCHAL_NUM_INTLEVELS`. That should cover irqs, debug and nmi all in one :) What do you think?